app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java

```diff
@@ -82,8 +82,15 @@ public SchemaContent getSchemaContent(int globalId) throws ArtifactNotFoundExcep
 
     @Override
     public Schema getSchema(String subject, String versionString) throws ArtifactNotFoundException, VersionNotFoundException, RegistryStorageException {
-        return parseVersionString(subject, versionString,
+        Schema schema = parseVersionString(subject, versionString,
                 version -> FacadeConverter.convert(subject, storage.getArtifactVersion(subject, version)));
+        
+        ArtifactMetaDataDto amd = storage.getArtifactMetaData(subject, ContentHandle.create(schema.getSchema()));
+        if(amd.getState().equals(ArtifactState.DELETED) || amd.getState().equals(ArtifactState.DISABLED)) {
```

Comment:
Do we really need this check here?
Imo, this should be in Storage itself / already.
e.g. so that storage never returns deleted or disabled artifact